### PR TITLE
integrate @sigstore/bundle into client

### PR DESCRIPTION
**Pull request by bdehamer**
#### Summary
Integrates the new `@sigstore/bundle` package into the `client` package. This PR is mostly just changes to import statements and deletion of the code which now lives in the `bundle` package.

### CODE REVIEW

File: packages/client/src/__tests__/tlog/verify/set.test.ts [hunk 1]
```diff
@@ -101,6 +101,23 @@ describe('verifyTLogSET', () => {
...
+              validFor: { start: undefined },
+            },
+          },
+        ];
+
+        it('returns false', () => {
+          expect(verifyTLogSET(entry, tlogs)).toBe(false);
+        });
+      });
+
```
Comment (bdehamer) [inline packages/client/src/__tests__/tlog/verify/set.test.ts:120]: Covers a scenario (a key in the `trusted_root.json` which was missing the actual bytes of the key) which was not previously tested.

File: packages/client/src/tlog/verify/checkpoint.ts [hunk 2]
```diff
@@ -196,36 +188,38 @@ class LogCheckpoint {
...
-): ValidTransparencyLogInstance[] {
+): sigstore.ViableTransparencyLogInstance[] {
   const targetDate = new Date(Number(integratedTime) * 1000);
-  return tlogInstances.filter((tlog): tlog is ValidTransparencyLogInstance => {
-    // Must have a log ID
-    if (!tlog.logId) {
-      return false;
-    }
+  return tlogInstances.filter(
+    (tlog): tlog is sigstore.ViableTransparencyLogInstance => {
```
Comment (bdehamer) [inline packages/client/src/tlog/verify/checkpoint.ts:194]: This is a neat trick . . . specifying a type assertion in the return value of the anonymous function called by `Array.filter` allows us to narrow the type of the return value.  We call `filter` on an array of `TransparencyLogInstance`s but the returned array contains `ViableTransparencyLogInstance`s (i.e. instances where we've validated that the required log ID and public key are present)

File: packages/client/src/tlog/verify/index.ts [hunk 3]
```diff
@@ -50,16 +56,11 @@ export function verifyTLogEntries(
...
+  entry: TransparencyLogEntry,
+  bundleContent: Bundle['content'],
   tlogs: sigstore.TransparencyLogInstance[],
   signingCert?: x509Certificate
 ): boolean {
-  // Check that the TLog entry has the fields necessary for verification
-  if (!sigstore.isVerifiableTransparencyLogEntry(entry)) {
-    return false;
-  }
-
```
Comment (bdehamer) [inline packages/client/src/tlog/verify/index.ts:62]: This check is now happening as part of the bundle validation process.

File: packages/client/src/tlog/verify/set.ts [hunk 4]
```diff
@@ -41,10 +43,6 @@ export function verifyTLogSET(
 
   // Check to see if we can verify the SET against any of the valid tlogs
   return validTLogs.some((tlog) => {
-    if (!tlog.publicKey?.rawBytes) {
-      return false;
-    }
-
```
Comment (bdehamer) [inline packages/client/src/tlog/verify/set.ts:47]: This check happens as part of the filter function above now

File: packages/client/package.json [hunk 5]
```diff
@@ -36,6 +36,7 @@
     "@types/make-fetch-happen": "^10.0.0"
   },
   "dependencies": {
+    "@sigstore/bundle": "^0.0.0",
```
Comment (feelepxyz) [inline packages/client/package.json:39]: Do we not want to publish an actual version of this package? Or follow up once we publish sigstore-js?

Comment (bdehamer) [reply packages/client/package.json:39]: Yeah, I'll be bumping to 1.0 before I cut a release with any of this stuff

**Pull request merged**